### PR TITLE
BUG: Series construction sharing data with DTI

### LINKED ISSUE #21907: pd.Series(dti) shares data with dti --> DatetimeIndex mutated
Appears to be specific to tz-aware

```
dti = pd.date_range('2016-01-01', periods=10, tz='US/Pacific')
ser = pd.Series(dti)
ser[::3] = pd.NaT

>>> dti
DatetimeIndex([                      'NaT', '2016-01-02 00:00:00-08:00',
               '2016-01-03 00:00:00-08:00',                       'NaT',
               '2016-01-05 00:00:00-08:00', '2016-01-06 00:00:00-08:00',
                                     'NaT', '2016-01-08 00:00:00-08:00',
               '2016-01-09 00:00:00-08:00',                       'NaT'],
              dtype='datetime64[ns, US/Pacific]', freq='D')
```

### PR DESCRIPTION
**Pull request by makbigc**
add a type check for DatetimeIndex and then pass a deep copy to create a Series object

closes #21907

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -434,7 +434,7 @@ Indexing
 - Fixed ``DataFrame[np.nan]`` when columns are non-unique (:issue:`21428`)
 - Bug when indexing :class:`DatetimeIndex` with nanosecond resolution dates and timezones (:issue:`11679`)
 - Bug where indexing with a Numpy array containing negative values would mutate the indexer (:issue:`21867`)
-
+- Bug when :class:`Series` was created from :class:`DatetimeIndex`, the series shares the same underneath data with that index (:issue:`21907`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: don't need the word underneath.

File: pandas/core/series.py [hunk 2]
```diff
@@ -206,7 +206,11 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                     data = data.astype(dtype)
                 else:
                     # need to copy to avoid aliasing issues
-                    data = data._values.copy()
```
Comment (jreback) [inline pandas/core/series.py]: wouldn't passing deep=True simply solve this?

**Review by jreback — CHANGES_REQUESTED** [review]
needs a test

**Comment by jreback** [comment]
also pls for all indices via the indices fixture, we DO test construction on all indices, so maybe you can just add onto these tests, look in pandas/tests/series/test_constructor.py

File: pandas/core/series.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/series.py]: I think the problem only occurs with tz-aware cases.

**Comment by jreback** [comment]
can you rebase and update

File: pandas/core/series.py (hunk 2 above)
Comment (jreback) [reply pandas/core/series.py]: this is not a good solution here, to special case things. pass ``deep=True`` here; if this doesn't work, then the copy constructor is broken.

Comment (makbigc) [reply pandas/core/series.py]: This bug is caused by the return value of _values. In datetimes.py, _values returns the DatetimeIndex itself for tz-aware and ndarray for tz-naive.
```
    @property
    def _values(self):
        # tz-naive -> ndarray
        # tz-aware -> DatetimeIndex
        if self.tz is not None:
            return self
        else:
            return self.values

```
That 's why  this problem only occurs with tz-aware cases as jbrockmendel said.
```

In [2]: dti1 = pd.date_range('2016-01-01', periods=10, tz='US/Pacific')

In [3]: dti2 = pd.date_range('2016-01-01', periods=10)

In [4]: ser1 = pd.Series(dti1)

In [5]: ser2 = pd.Series(dti2)

In [6]: ser1[::3]=pd.NaT

In [7]: ser2[::3]=pd.NaT

In [8]: dti1
Out[8]: 
DatetimeIndex([                      'NaT', '2016-01-02 00:00:00-08:00',
               '2016-01-03 00:00:00-08:00',                       'NaT',
               '2016-01-05 00:00:00-08:00', '2016-01-06 00:00:00-08:00',
                                     'NaT', '2016-01-08 00:00:00-08:00',
               '2016-01-09 00:00:00-08:00',                       'NaT'],
              dtype='datetime64[ns, US/Pacific]', freq='D')

In [9]: dti2
Out[9]: 
DatetimeIndex(['2016-01-01', '2016-01-02', '2016-01-03', '2016-01-04',
               '2016-01-05', '2016-01-06', '2016-01-07', '2016-01-08',
               '2016-01-09', '2016-01-10'],
              dtype='datetime64[ns]', freq='D')

```

**Comment by jreback** [comment]
closing as superseded by #24096

**Pull request closed without merging**